### PR TITLE
a8n: Fetch and parse CheckSuite status from GitHub

### PR DESCRIPTION
**Pull request by ryanslade**

### CODE REVIEW

File: cmd/repo-updater/repos/testdata/golden/GithubSource_CreateChangeset_already exists [hunk 1]
```diff
@@ -35,25 +35,25 @@
     "Item": {
```
Comment (unknwon) [inline cmd/repo-updater/repos/testdata/golden/GithubSource_CreateChangeset_already exists]: This file name contains a space! :trollface: Should this happen?

Comment (ryanslade) [reply cmd/repo-updater/repos/testdata/golden/GithubSource_CreateChangeset_already exists]: I'm not sure, but it has been that way for a while and hasn't broken anything 🤷‍♂

Comment (unknwon) [reply cmd/repo-updater/repos/testdata/golden/GithubSource_CreateChangeset_already exists]: There is another file few sections below, that actually ends with `already-exists`, something must be inconsistent 😄

Comment (mrnugget) [reply cmd/repo-updater/repos/testdata/golden/GithubSource_CreateChangeset_already exists]: Yep, I forgot to add the `strings.Replace` call to this line here: https://github.com/sourcegraph/sourcegraph/blob/d5d27322c52eed24b04f2386610fab8d87abb39f/cmd/repo-updater/repos/github_test.go#L77

Comment (ryanslade) [reply cmd/repo-updater/repos/testdata/golden/GithubSource_CreateChangeset_already exists]: Ah, cool, fix incoming

File: internal/extsvc/github/client.go [hunk 2]
```diff
@@ -310,6 +310,10 @@ func (c *Client) requestGraphQL(ctx context.Context, token, query string, vars m
 	if err != nil {
 		return err
 	}
+
+	// Enable Checks API
+	// https://developer.github.com/v4/previews/#checks
+	req.Header.Add("Accept", "application/vnd.github.antiope-preview+json")
```
Comment (mrnugget) [inline internal/extsvc/github/client.go]: Just to be sure: does this add another value if we already have an `Accept` header ([see the last example here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept#Examples) or does this overwrite the existing `Accept` header? I think we want to be sure to not overwrite something.

File: internal/extsvc/github/pulls.go [hunk 3]
```diff
@@ -698,6 +717,18 @@ fragment commit on Commit {
...
+    avatarUrl
+    email
+    name
+    user {
+      ...actor
+    }
+  }
+}
+
+fragment basicCommit on Commit {
```
Comment (mrnugget) [inline internal/extsvc/github/pulls.go]: Hmmm, isn't the naming reversed? As in: this is the pull request commit, if I understand correctly, and a basic commit would be a normal commit?

File: internal/extsvc/github/client.go (hunk 2 above)
Comment (ryanslade) [reply internal/extsvc/github/client.go]: It adds it, a `Header` is just a `map[string][]string`

https://play.golang.org/p/VdmDTBTeBhF

File: internal/extsvc/github/pulls.go (hunk 3 above)
Comment (ryanslade) [reply internal/extsvc/github/pulls.go]: Ah, yes, good point. I modified that a few times. I'll fix it now

File: internal/extsvc/github/client.go (hunk 2 above)
Comment (mrnugget) [reply internal/extsvc/github/client.go]: Great!

**Comment by ryanslade** [comment]
Part of https://github.com/sourcegraph/sourcegraph/issues/7093

**Pull request merged**